### DISCUSSION TITLE
XMLA endpoint connection to Power BI + redshift

### DISCUSSION THREAD

**Discussion by rafaelsmoreno** [TabularEditor/TabularEditor3#330] [🙋♀️TE3 Q&A]
Hi, I've been wrapping my head around an issue and though maybe I could get some ideas from here. 

 I have the following scenario: a PBI dataset consuming data from AWS redshift data warehouse. 

I connected TE3 to the dataset through xmla endpoint connection, and all works fine. However, I couldn't add tables to the model, as I couldn't establish a connection to redshift (even though I have the odbc properly set). There isn't a option to insert redshift credentials, and as a result, nothing works. 

I might be doint something wrong, but I'd appreciate some guidance on that. 

Thanks in advance!

**Comment by otykier** [comment]
Hi @rafaelsmoreno 

Unfortunately, we don't support Redshift natively (we will add native support for more types of data sources over time, but don't expect us to reach parity with the 300+ data sources supported by Power BI any time soon). That being said, ODBC should work if you configure a local DSN (remember that your ODBC DSN platform should match TE3, that is either 32-bit (x86) or 64-bit).

Alternatively, you may rely on Analysis Services to import tables and perform schema updates as described [here](https://docs.tabulareditor.com/te3/tutorials/importing-tables.html#updating-table-schema-through-analysis-services). This way, it should be easy to import new tables if you already have a data model (created in Power BI Desktop), that contains at least one table from that data source.

If that fails, there is always the option of creating a new blank model in Desktop, import the tables you need using Desktop, and then finally use TE3 to copy the newly created tables over to the dataset you're working on through XMLA.

**Reply by rafaelsmoreno** [reply]
I don't have anything on azure or analysis services, it's 100% at AWS. The idea was updating the datamodel at pbi service through xmla endpoint exactly because the pbix is huge by now, and will get bigger. From your answer the idea would be merging a pbit from a pbix with only the tables to be added into the existing/main dataset, is that so? 

Regarding the odbc, I'm failing at finding where exactly to insert my credentials: 
![image](https://user-images.githubusercontent.com/30775503/147125791-9e60b984-40d3-405e-a2d2-6fa04d1df3b7.png)
